src/Microsoft.Windows.CsWin32/Generator.MetadataHelpers.cs

```diff
@@ -484,9 +502,13 @@ private bool IsManagedType(TypeDefinitionHandle typeDefinitionHandle)
 
                 throw new NotSupportedException();
             }
+            catch (GenerationFailedException)
+            {
+                throw;
+            }
```

Comment:
🤔 Why avoid wrapping the exception with higher level info?

Reply:
There are tests that are intentionally causing an exception and if we fall into the other catch block then we call ToTypeSyntax and immediately trigger the same exception and stack overflow. This enabled us to take the original GenerationFailedException and just bubble it out instead of unwrapping it and trying to re-generate the exact same message again (but failing).

Reply:
If you can suggest a safer way to generate this exception message without potentially causing the exact same exception while doing so, I'm open to suggestions. :)